src/app/search-page/search-result.component/search-result.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { Router } from "@angular/router";
+import { IProduct } from "src/app/Interface/products";
+
+@Component({
+  selector: 'app-search-result',
+  templateUrl: './search-result.component.html',
+  styleUrls: [],
+})
+
+export class SearchResultComponent implements OnInit {
+
+  @Input() visibleProducts: IProduct[] = [];
+
+  breakPoint!: number;
+
+  constructor(private router: Router) {}
+
+  onResize(event: any) {
+    this.breakPoint = (event.target.innerWidth < 1200) ? 2 : 3;
+  }
+
+  toProductDetailPage(id: number): void {
+    this.router.navigate(["product", id]);
+  }
+
+  ngOnInit(): void {
+    this.breakPoint = (window.innerWidth < 1200) ? 2 : 3;
+  }
```

Comment:
onResize and ngOnInit have almost the same content.
Could you merge this into one function? :)

Reply:
I've tried, but I do not understand how to merge it into one function, if the first function takes 'event' as a parameter and other function will be called by ngOnInit and has 'window' parameter.